lib/syskit/task_context.rb

```diff
@@ -419,6 +442,25 @@ def commit_properties_if_needed(*args)
             end
         end
 
+        def start!
+            return super unless read_only?
+
+            event(:start).emit
+        end
```

Comment:
Oh my.

Just return from the block in `event :start do |context| ... end`

This is very wrong.
- Calling an event can be done is more ways than calling `start!`
- it bypasses the checks
- it emits the event even if the remote task is not running